File: frontend/src/types/report.ts
export type ReportTargetType = "PRODUCT" | "COMPANY" | "REVIEW" | "USER";

export type ReportReason =
  | "SPAM"
  | "OFFENSIVE_CONTENT"
  | "MISLEADING_INFO"
  | "COUNTERFEIT"
  | "HARASSMENT"
  | "FRAUD"
  | "COPYRIGHT_VIOLATION"
  | "OFF_TOPIC"
  | "FAKE"
  | "OTHER";

export const REPORT_STATUSES = {
  OPEN: "Open",
  ACTIONED: "Actioned",
  DISMISSED: "Dismissed",
} as const;

export type ReportStatus = keyof typeof REPORT_STATUSES;

export interface ReportScreenshot {
  id: string;
  url: string;
  position: number;
}

export interface Report {
  id: string;
  targetType: ReportTargetType;
  targetId: string;
  reporterId: string;
  reason: ReportReason;
  title: string;
  description: string;
  status: ReportStatus;
  screenshots: ReportScreenshot[];
  createdAt: string;
  resolvedAt: string | null;
  resolvedBy: string | null;
  targetName: string | null;
  reporterName: string | null;
}

File: backend/src/main/java/backend/models/core/Report.java
package backend.models.core;

import backend.models.enums.ReportReason;
import backend.models.enums.ReportStatus;
import backend.models.enums.ReportTargetType;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.Instant;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

@Entity
@Getter
@Setter
@NoArgsConstructor
@Table(
        name = "reports",
        uniqueConstraints = @UniqueConstraint(
                name = "uq_report_target_reporter",
                columnNames = {"target_type", "target_id", "reporter_id"}
        ),
        indexes = {
                @Index(name = "idx_report_status", columnList = "status, created_at"),
                @Index(name = "idx_report_target", columnList = "target_type, target_id")
        }
)
@EntityListeners(AuditingEntityListener.class)
public class Report {

    @Id
    @org.hibernate.annotations.UuidGenerator(style = org.hibernate.annotations.UuidGenerator.Style.TIME)
    private UUID id;

    @Enumerated(EnumType.STRING)
    @Column(name = "target_type", nullable = false, length = 20)
    private ReportTargetType targetType;

    @Column(name = "target_id", nullable = false)
    private UUID targetId;

    @Column(name = "reporter_id", nullable = false)
    private UUID reporterId;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 40)
    private ReportReason reason;

    @Column(nullable = false, length = 150)
    private String title;

    @Column(nullable = false, length = 2000)
    private String description;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private ReportStatus status = ReportStatus.OPEN;

    @OneToMany(mappedBy = "report", cascade = CascadeType.ALL, orphanRemoval = true)
    @OrderBy("position ASC")
    private List<ReportScreenshot> screenshots = new ArrayList<>();

    @CreatedDate
    @Column(nullable = false, updatable = false)
    private Instant createdAt;

    @Column(name = "resolved_at")
    private Instant resolvedAt;

    @Column(name = "resolved_by")
    private UUID resolvedBy;
}
